src/TheThingsNetwork.cpp

```diff
@@ -282,7 +282,13 @@ int TheThingsNetwork::sendBytes(const byte* payload, int length, int port, bool
     return -1;
   }
 
-  String response = readLine(10000);
+  String response = "";
+  if (confirm == true) {
+    response = readLine(100000);
```

Comment:
Wait indefinitely; even 100 seconds is not enough with confirm, high retry count, duty cycle limitation and high spreading factor.